server/api/src/rsExecRuleExpression.cpp

```diff
@@ -1,77 +1,59 @@
+#include "rsExecRuleExpression.hpp"
+
 #include "irods_re_plugin.hpp"
 #include "miscServerFunct.hpp"
 #include "rcMisc.h"
 #include "packStruct.h"
-#include "rsExecRuleExpression.hpp"
 
+#include <cstring>
 #include <string>
 
 extern const packInstruct_t RodsPackTable[];
 
-int rsExecRuleExpression(
-    rsComm_t*               _comm,
-    exec_rule_expression_t* _exec_rule ) {
-    if( !_comm || !_exec_rule ) {
+int rsExecRuleExpression(RsComm* _comm, ExecRuleExpression* _exec_rule)
+{
+    if (!_comm || !_exec_rule) {
         return SYS_INVALID_INPUT_PARAM;
     }
 
-    std::string my_rule_text = (char*)_exec_rule->rule_text_.buf;
-
-    ruleExecInfoAndArg_t* rei_and_arg = NULL;
-    int status = unpackStruct(
-                     _exec_rule->packed_rei_.buf,
-                     ( void ** ) &rei_and_arg,
-                     "ReiAndArg_PI",
-                     RodsPackTable,
-                     NATIVE_PROT );
-    if ( status < 0 ) {
-        rodsLog(
-            LOG_ERROR,
-            "%s :: unpackStruct error. status = %d",
-            __FUNCTION__,
-            status );
+    ruleExecInfoAndArg_t* rei_and_arg = nullptr;
+    const auto status = unpackStruct(_exec_rule->packed_rei_.buf,
+                                     reinterpret_cast<void **>(&rei_and_arg),
+                                     "ReiAndArg_PI",
+                                     RodsPackTable,
+                                     NATIVE_PROT);
+    if (status < 0) {
+        rodsLog(LOG_ERROR, "%s :: unpackStruct error. status = %d", __FUNCTION__, status);
         return status;
     }
 
     ruleExecInfo_t* rei = rei_and_arg->rei;
     rei->rsComm = _comm;
     rei->rsComm->clientUser = *rei->uoic;
 
-    // do doi things?
-    if ( rei->doi != NULL ) {
-        if ( rei->doi->next != NULL ) {
-            free( rei->doi->next );
-            rei->doi->next = NULL;
+    // Do dataObjectInfo (doi) things?
+    if (rei->doi) {
+        if (rei->doi->next) {
+            free(rei->doi->next);
+            rei->doi->next = nullptr;
         }
     }
 
     std::string instance_name;
-    if(strlen(rei->pluginInstanceName) > 0) {
+    if (std::strlen(rei->pluginInstanceName) > 0) {
```

Comment:
What happens if the string is not null terminated? This needs to use a safer function to check for empty.